### PR TITLE
feature: support user-supplied bucket name prefix

### PR DESCRIPTION
**Pull request by adlersantos**
## Description

This PR implements a `--bucket-name-prefix` parameter for `scripts/generate_terraform.py`. The previous implementation of using the GCP project number has been removed.

The benefits of using a user-supplied bucket name prefix are as follows:
* Prevents GCS bucket naming collisions
* Better readability and memorize-ability for data consumers
* Make naming collisions easy to troubleshoot: If a collision is encountered for a specific env (say, prod), one can simply change the bucket name prefix for that stage

## Checklist
- [x] Tests pass.
- [x] Linters pass.
- [x] Please merge this PR for me once it is approved.
- [x] If this PR adds/edits/deletes a feature, I have updated the [`README`](https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/main/README.md) accordingly.
- [x] If this PR adds/edits a dataset or pipeline, it was reviewed and approved by the Google Cloud Public Datasets team beforehand.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,6 +190,7 @@ Every dataset and pipeline folder must contain a `dataset.yaml` and a `pipeline.
 
 # Best Practices
 
+- When running `scripts/generate_terraform.py`, the argument `--bucket-name-prefix` helps prevent GCS bucket name collisions because bucket names must be globally unique. Make the prefix as unique as possible, and specific to your own environment or use case.
```
Comment (leahecole) [inline README.md:193]: There needs to be some mention about this in step 3 of the instructions as well

**Review by leahecole — CHANGES_REQUESTED** [review]
Minor change about updating the README

then my only question is are the other changes to the datasets themselves autogenerated? Asking because I noticed some inconsistencies in the before and after with hyphens and underscores in the names (`vizgen-merfish` vs `vizgen_merfish`)

File: README.md (hunk 1 above)
Comment (adlersantos) [reply README.md:196]: Good catch! Please see 8c27336.

**Comment by adlersantos** [comment]
> Minor change about updating the README
> 
> then my only question is are the other changes to the datasets themselves autogenerated? Asking because I noticed some inconsistencies in the before and after with hyphens and underscores in the names (`vizgen-merfish` vs `vizgen_merfish`)

Yes, the `.tf` files are all auto-generated. 

And yes, you're right in noticing that. I noticed that we're using a mix of underscores and hyphens for the generate bucket names, previously `{PREFIX}-vizgen_merfish` and I thought we're renaming the bucket anyway, so I might as well set a standard to use hyphens instead of underscores. 

On that note, I added that convention here: 03ad1aa

**Comment by adlersantos** [comment]
@leahecole I also added tests for the hyphenated bucket naming convention: 16818df and bfebdd3

**Review by leahecole — APPROVED** [review]
LGTM! I tried to break it locally and was so satisfied at the error! :)

**Pull request merged**